<html>
    <head>
        <meta property="og:image" content="https://grcodeclub.github.io/file/background/3.png"/>
        <meta property="og:title" content="Γλώσσα Προγραμματισμού C++ | Χειρισμός εξαιρέσεων">
        <script src="https://grcodeclub.github.io/js/add/head.js"></script>
        <script src="https://grcodeclub.github.io/programming/js/menu.js"></script>
    </head>

     <body class="programming">
        <div class="container">
            <div id="add_headder_and_menu"></div><script src="https://grcodeclub.github.io/js/add/headder_and_menu.js"></script>

            <div class="Text"> <!--Αρχή Κειμένου-->
                <a href="https://grcodeclub.gr/cpp"><h1 id="back_page"></h1></a> <script src="https://grcodeclub.github.io/js/add/back_page.js"></script>
                <div class="text_bg"> <!-- Αρχή Background Κειμένου-->

                    <h1 class="title">Χειρισμός εξαιρέσεων</h1>
                    <p>Κατά τη διάρκεια ανάπτυξης ενός προγράμματος, μπορεί να υπάρξουν περιπτώσεις, όπου δεν έχουμε τη βεβαιότητα ότι το συγκεκριμένο κομμάτι κώδικα θα δουλέψει σωστά, είτε επειδή μπορεί να προσπαθήσει να προσπελάσει πόρους που δεν υπάρχουν, είτε επειδή βγαίνει εκτός ορίων, ...</p>
                    <div class="bg_preview">
                    <div  class="line-numbers"><pre class="code_editor"><code class="language-cpp">try { // κώδικας που θα δοκιμαστεί
	throw exception;}
catch (type  exception) {
	// κώδικας που θα εκτελεστεί σε περίπτωση εξαίρεσης
}</code></pre></div></div>
                <ol>
                    <li>Ο κώδικας στο μπλοκ try εκτελείται κανονικά</li>
                    <li>Αν κάτι συμβεί ο κώδικας χρησιμοποιεί τη λέξη throw και μια παράμετρο για να προκαλέσει εξαίρεση</li>
                    <li>Αν έχει συμβεί μια εξαίρεση, το μπλοκ catch εκτελείται λαμβάνοντας ως παράμετρο αυτή που περάστηκε από το throw</li>
                </ol>

                <h3 class="highlight_black"></h3>
                <p>Η C++ διαθέτει προκαθορισμένες κλάσεις εξαιρέσεων στην std βιβλιοθήκη. 
                    Η std::exception είναι η βασική κλάση από την οποία μπορούν να κληρονομηθούν 
                    άλλες κλάσεις εξαιρέσεων.</p>
        
                </div><!-- Τέλος Background Κειμένου-->

            </div> <!--Τέλος Κειμένου-->

        </div> <!--Τέλος container-->
        
        <!-- Footer -->
        <footer id="add-social"></footer><script src="https://grcodeclub.github.io/js/add/social.js"></script>
    </body>
</html>
